### PR TITLE
fix: organization apis are accessible through api key

### PR DESCRIPTION
**Pull request by jainpawan21**
### What change does this PR introduce?
1. Remove `@ApiExcludeController()` and add `@ExternalApiAccessible()` decorator
2. Add `OrganizationResponseDto`

Fixes NV-1920

### Why was this change needed?
So that organizations' APIs can be accessed using API Key

### CODE REVIEW

File: apps/api/src/app/organization/organization.controller.ts [hunk 1]
```diff
@@ -120,6 +147,10 @@ export class OrganizationController {
   }
 
   @Get('/members')
+  @ExternalApiAccessible()
+  @ApiOperation({
+    summary: 'Fetch all members of current organizations',
+  })
   async getMember(@UserSession() user: IJwtPayload) {
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/organization.controller.ts:154]: Do we need to add here?
@ApiResponse(MemberDto, 200, true)

File: apps/api/src/app/organization/organization.controller.ts [hunk 2]
```diff
@@ -158,7 +182,12 @@ export class OrganizationController {
   }
 
   @Patch('/')
+  @ExternalApiAccessible()
   @Roles(MemberRoleEnum.ADMIN)
+  @ApiResponse(RenameOrganizationDto)
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/organization.controller.ts:187]: Should be with 200?


```suggestion
  @ApiResponse(RenameOrganizationDto, 200)
```

File: packages/node/README.md
Comment (djabarovgeorge) [inline packages/node/README.md:1]: 👏

File: apps/api/src/app/organization/organization.controller.ts (hunk 2 above)
Comment (jainpawan21) [reply apps/api/src/app/organization/organization.controller.ts:190]: @ApiResponse() has 200 as default status code

File: apps/api/src/app/organization/dtos/organization-response.dto.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  direction?: DirectionEnum;
+}
+
+export class OrganizationResponseDto {
+  @IsString()
+  name: string;
+
+  @IsString()
+  @IsOptional()
+  logo?: string;
```
[the remark this answers is not in the document]
Comment (jainpawan21) [reply apps/api/src/app/organization/dtos/organization-response.dto.ts:42]: Added ✅

**Comment by jainpawan21** [comment]
@djabarovgeorge Please look into this PR again 🙂

File: apps/api/src/app/organization/organization.controller.ts (hunk 2 above)
Comment (djabarovgeorge) [reply apps/api/src/app/organization/organization.controller.ts:190]: I see, good to know :)

File: apps/api/src/app/organization/dtos/member-response.dto.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  @ApiProperty()
+  @IsString()
+  token: string;
+
+  @ApiProperty()
+  @IsDate()
+  invitationDate: Date;
+
+  @ApiProperty()
+  @IsDate()
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/member-response.dto.ts:37]:
```suggestion
  @IsDate()
  @ApiPropertyOptional()
```

File: apps/api/src/app/organization/dtos/member-response.dto.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  answerDate?: Date;
+
+  @ApiProperty()
+  @IsDate()
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/member-response.dto.ts:41]:
```suggestion
  @IsString()
  ```

File: apps/api/src/app/organization/dtos/member-response.dto.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+  @ApiPropertyOptional()
+  @IsObject()
+  user?: MemberUserDto;
+
+  @ApiPropertyOptional({
+    enum: MemberRoleEnum,
+    isArray: true,
+  })
+  @IsArray()
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/member-response.dto.ts:62]:
```suggestion
  @IsEnum(MemberRoleEnum)
```

File: apps/api/src/app/organization/dtos/member-response.dto.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  roles?: MemberRoleEnum;
+
+  @ApiPropertyOptional()
+  @IsObject()
+  invite?: MemberInviteDTO;
+
+  @ApiPropertyOptional({
+    enum: { ...MemberStatusEnum },
+  })
+  @IsString()
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/member-response.dto.ts:72]:
```suggestion
  @IsEnum(MemberStatusEnum)
```

File: apps/api/src/app/organization/dtos/organization-response.dto.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import { PartnerTypeEnum, DirectionEnum } from '@novu/dal';
+import { IsObject, IsOptional, IsString } from 'class-validator';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { UpdateBrandingDetailsDto } from './update-branding-details.dto';
+
+export class IPartnerConfigurationResponseDto {
+  @ApiPropertyOptional()
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/organization-response.dto.ts:7]:
```suggestion
  @ApiPropertyOptional()
  @IsArray()
  @IsString({ each: true })
```

File: apps/api/src/app/organization/dtos/organization-response.dto.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  projectIds?: string[];
+
+  @ApiProperty()
+  accessToken: string;
+
+  @ApiProperty()
+  configurationId: string;
+
+  @ApiPropertyOptional()
+  teamId: string;
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/organization-response.dto.ts:17]: @IsString()

File: apps/api/src/app/organization/dtos/organization-response.dto.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+  @ApiProperty({
+    enum: { ...PartnerTypeEnum },
+    description: 'Partner Type Enum',
+  })
+  @IsString()
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/organization-response.dto.ts:23]:
```suggestion
  @IsEnum(PartnerTypeEnum)
```

File: apps/api/src/app/organization/dtos/organization-response.dto.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+  @ApiPropertyOptional()
+  @IsString()
+  logo?: string;
+
+  @ApiProperty()
+  @IsObject()
+  branding: OrganizationBrandingResponseDto;
+
+  @ApiPropertyOptional()
```
Comment (djabarovgeorge) [inline apps/api/src/app/organization/dtos/organization-response.dto.ts:48]:
```suggestion
  @ApiPropertyOptional()
   @IsObject()
```

**Review by djabarovgeorge — APPROVED** [review]
Yeah looks good to me, left a couple of comments related to class-validator and nestjs/swagger decorators.
To be honest, I don't even know if we need class-validator decorators in the DTO because we don't validate them, it could be a good idea to add validation IMO but from what i know we still don't.

**Pull request merged**